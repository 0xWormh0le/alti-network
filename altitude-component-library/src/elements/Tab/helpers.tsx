export const getTabIndexOf = (element: HTMLElement) => {
  const tabs = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tab') as HTMLElement[]
  return tabs.indexOf(element)
}

export const getTabPanelIndexOf = (element: HTMLElement) => {
  const tabPanels = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tabPanel') as HTMLElement[]
  return tabPanels.indexOf(element)
}
