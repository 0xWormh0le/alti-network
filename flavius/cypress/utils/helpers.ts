export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
