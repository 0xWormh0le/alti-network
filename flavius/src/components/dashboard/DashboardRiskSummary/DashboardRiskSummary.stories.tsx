import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DashboardRiskSummary from './DashboardRiskSummary'

export default {
  title: 'Dashboard/DashboardRiskSummary',
  component: DashboardRiskSummary
} as Meta

const Template: Story<any> = (args) => <DashboardRiskSummary {...args} />

export const Default = Template.bind({})
Default.args = {}
