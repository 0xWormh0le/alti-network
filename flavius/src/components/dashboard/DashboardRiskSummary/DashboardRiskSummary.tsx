import React, { useState, useEffect } from 'react'
import API from '@aws-amplify/api/lib'
import { GENERAL_URLS } from 'api/endpoints'
import DashboardGrid from '../DashboardGrid'
import { alertError } from 'util/alert'
import { DashboardCard } from '../DashboardCards'
import { useCancelablePromise } from 'util/hooks'
import { riskTypeGroups } from 'config'

const INITIAL_CARDS: DashboardCard[] = riskTypeGroups.map((item) => ({
  groupType: item.groupType,
  label: item.name,
  value: 0
}))

// TODO: This method can utilize the consts defined in Models/RiskCatalog.ts
// Returns true if the risk type id is within this card group, otherwise, returns false
const riskBelongsToCardGroup = (riskTypeId: number, card: DashboardCard | undefined) => {
  if (!card) {
    return false
  }

  const lowerBound = card.groupType * 1000
  const upperBound = lowerBound + 1000

  return riskTypeId < upperBound && riskTypeId >= lowerBound
}

// Return the updated cards with the count for each category
const addCountToCards = (cards: DashboardCard[], stats: RiskTypeSummary[]) => {
  return stats.reduce(
    (updatedCards, riskType) => {
      const riskTypeId = riskType.riskTypeId
      return updatedCards.map((card) => {
        if (riskBelongsToCardGroup(riskTypeId, card)) {
          return { ...card, value: (card.value += riskType.count) }
        } else {
          return card
        }
      })
    },
    cards.map((card) => ({ ...card, value: 0 }))
  ) // reset count in cards before computing sum
}

const DashboardRiskSummary: React.FC<{}> = () => {
  const [loading, setLoading] = useState(true)
  const [cards, setCards] = useState<DashboardCard[]>(INITIAL_CARDS)
  const [selectedCardGroupType, setSelectedCardGroupType] = useState(1)
  const [riskTypes, setRiskTypes] = useState<RiskTypeSummary[]>([])
  const cancelablePromise = useCancelablePromise()

  useEffect(() => {
    const fetchStats = async () => {
      setLoading(true)

      try {
        const apiResponse = await cancelablePromise(API.get('risks', `${GENERAL_URLS.RISKS}/stats`, {}))
        const updatedCards = addCountToCards(INITIAL_CARDS, apiResponse.stats)

        setCards(updatedCards)
        setRiskTypes(apiResponse.stats)
        setLoading(false)
      } catch (error) {
        if (error.name !== 'PromiseCanceledError') {
          alertError(error.message)
          setLoading(false)
        }
      }
    }

    fetchStats()
  }, [cancelablePromise])

  const selectedCard = cards.find((card) => selectedCardGroupType === card.groupType)
  const riskTypesToRender = riskTypes.filter((riskType) => riskBelongsToCardGroup(riskType.riskTypeId, selectedCard))

  return (
    <div className='DashboardRiskSummary'>
      <DashboardGrid
        loading={loading}
        cards={cards}
        selectedCardGroupType={selectedCardGroupType}
        onCardClick={(cardGroupType: number) => setSelectedCardGroupType(cardGroupType)}
        riskTypes={riskTypesToRender}
      />
    </div>
  )
}

export default DashboardRiskSummary
