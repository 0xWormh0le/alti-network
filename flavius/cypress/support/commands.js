import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
