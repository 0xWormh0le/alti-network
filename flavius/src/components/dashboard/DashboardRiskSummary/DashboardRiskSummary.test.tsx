import React from 'react'
import { fireEvent, waitForElement } from '@testing-library/react'
import { renderWithRouter } from 'test/support/helpers'
import API from '@aws-amplify/api/lib'
import DashboardRiskSummary from './DashboardRiskSummary'

jest.mock('@aws-amplify/api/lib')

beforeEach(() => {
  const apiGet: any = API.get
  apiGet.mockResolvedValue({
    stats: [
      { riskTypeId: 0, count: 22, severity: 5 as SeverityRange },
      { riskTypeId: 10, count: 4, severity: 5 as SeverityRange },

      { riskTypeId: 1011, count: 3, severity: 5 as SeverityRange },
      { riskTypeId: 1020, count: 27, severity: 8 as SeverityRange },
      { riskTypeId: 1021, count: 5, severity: 9 as SeverityRange },

      { riskTypeId: 2000, count: 15, severity: 6 as SeverityRange },
      { riskTypeId: 2001, count: 9, severity: 9 as SeverityRange },
      { riskTypeId: 2010, count: 5, severity: 9 as SeverityRange },
      { riskTypeId: 2011, count: 30, severity: 9 as SeverityRange },

      { riskTypeId: 3100, count: 73, severity: 6 as SeverityRange }
    ]
  })
})

describe('General rendering', () => {
  it('renders correctly', () => {
    const { container } = renderWithRouter(<DashboardRiskSummary />)
    expect(container).toMatchSnapshot()
  })
  it('renders cards, preview table and "Show all risks" button', async () => {
    const { getByText, getByLabelText } = renderWithRouter(<DashboardRiskSummary />)

    const sharingRisksCard = await waitForElement(() => getByText('Sharing Risks'))

    // cards
    expect(sharingRisksCard).toBeInTheDocument()
    expect(getByText('Relationship Risks')).toBeInTheDocument()
    expect(getByText('Activity Risks')).toBeInTheDocument()
    expect(getByText('Informational Risks')).toBeInTheDocument()

    // table
    expect(getByText('Risk severity')).toBeInTheDocument()
    expect(getByText('Risk count')).toBeInTheDocument()

    // show all risks button
    expect(getByLabelText('show all risks')).toBeInTheDocument()
  })
})

describe('Buttons', () => {
  let getAllByText: (...args: any[]) => any
  let getByLabelText: (...args: any[]) => any
  let getByText: (...args: any[]) => any

  beforeEach(() => {
    const renderResult = renderWithRouter(<DashboardRiskSummary />)
    getByLabelText = renderResult.getByLabelText
    getAllByText = renderResult.getAllByText
    getByText = renderResult.getByText
  })

  it('clicking "Show all risks" button redirects to Top Risks page', async () => {
    const showAllRisksButton = await waitForElement(() => getByLabelText('show all risks'))
    const anchor = showAllRisksButton.parentElement

    expect(anchor.getAttribute('href')).toBe('/risks?riskTypeIds=[1021,1020,1011]')
  })

  it('clicking the first "See all" button redirects to Top Risks page with filters applied', async () => {
    const firstSeeAllButton = await waitForElement(() => getAllByText('See all')[0])
    const anchor = firstSeeAllButton.parentElement

    expect(anchor.getAttribute('href')).toBe('/risks?riskTypeIds=[1021]')
  })

  it('shows label for show `all risks button` with correct category name when active card is changed', async () => {
    const showAllRisksButton = await waitForElement(() => getByLabelText('show all risks'))
    expect(showAllRisksButton).toHaveTextContent('Show All Sharing Risks')
    getByText('Relationship Risks').click()
    expect(showAllRisksButton).toHaveTextContent('Show All Relationship Risks')
    getByText('Activity Risks').click()
    expect(showAllRisksButton).toHaveTextContent('Show All Activity Risks')
    getByText('Informational Risks').click()
    expect(showAllRisksButton).toHaveTextContent('Show All Informational Risks')
  })
})

describe('Cards', () => {
  it('another cards gets selected when clicked', async () => {
    const { getByText, container } = renderWithRouter(<DashboardRiskSummary />)

    const generalRisksCard = await waitForElement(() => getByText('Informational Risks'))
    fireEvent.click(generalRisksCard)

    // check that the 4th card is now selected
    expect(container.querySelectorAll('.DashboardCards__card')[3]).toHaveClass('DashboardCards__card--active')
  })
})
