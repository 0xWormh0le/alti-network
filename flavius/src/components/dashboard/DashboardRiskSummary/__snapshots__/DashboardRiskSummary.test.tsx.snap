// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General rendering renders correctly 1`] = `
<div>
  <div
    class="DashboardRiskSummary"
  >
    <div
      class="DashboardGrid"
    >
      <h3
        class="Typography Typography--h3 SectionTitle"
      >
        Risk Summary
      </h3>
      <div
        class="DashboardGrid__main"
      >
        <div
          class="DashboardCards"
        >
          <div
            class="DashboardCards__card DashboardCards__card--active"
          >
            <svg
              aria-labelledby="DashboardCards__card-value-aria"
              class="DashboardCards__card-value"
              height="90"
              role="img"
            >
              <title
                id="DashboardCards__card-value-aria"
              >
                Loading...
              </title>
              <rect
                clip-path="url(#DashboardCards__card-value-diff)"
                height="100%"
                role="presentation"
                style="fill: url(#DashboardCards__card-value-animated-diff);"
                width="100%"
                x="0"
                y="0"
              />
              <defs
                role="presentation"
              >
                <clippath
                  id="DashboardCards__card-value-diff"
                >
                  <rect
                    height="90"
                    width="100%"
                    x="0"
                    y="0"
                  />
                </clippath>
                <lineargradient
                  id="DashboardCards__card-value-animated-diff"
                >
                  <stop
                    offset="0%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-2; -2; 1"
                    />
                  </stop>
                  <stop
                    offset="50%"
                    stop-color="#e5f3fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-1; -1; 2"
                    />
                  </stop>
                  <stop
                    offset="100%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="0; 0; 3"
                    />
                  </stop>
                </lineargradient>
              </defs>
            </svg>
            <span
              class="Typography Typography--label DashboardCards__card-label"
            >
              Sharing Risks
            </span>
          </div>
          <div
            class="DashboardCards__card"
          >
            <svg
              aria-labelledby="DashboardCards__card-value-aria"
              class="DashboardCards__card-value"
              height="90"
              role="img"
            >
              <title
                id="DashboardCards__card-value-aria"
              >
                Loading...
              </title>
              <rect
                clip-path="url(#DashboardCards__card-value-diff)"
                height="100%"
                role="presentation"
                style="fill: url(#DashboardCards__card-value-animated-diff);"
                width="100%"
                x="0"
                y="0"
              />
              <defs
                role="presentation"
              >
                <clippath
                  id="DashboardCards__card-value-diff"
                >
                  <rect
                    height="90"
                    width="100%"
                    x="0"
                    y="0"
                  />
                </clippath>
                <lineargradient
                  id="DashboardCards__card-value-animated-diff"
                >
                  <stop
                    offset="0%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-2; -2; 1"
                    />
                  </stop>
                  <stop
                    offset="50%"
                    stop-color="#e5f3fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-1; -1; 2"
                    />
                  </stop>
                  <stop
                    offset="100%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="0; 0; 3"
                    />
                  </stop>
                </lineargradient>
              </defs>
            </svg>
            <span
              class="Typography Typography--label DashboardCards__card-label"
            >
              Relationship Risks
            </span>
          </div>
          <div
            class="DashboardCards__card"
          >
            <svg
              aria-labelledby="DashboardCards__card-value-aria"
              class="DashboardCards__card-value"
              height="90"
              role="img"
            >
              <title
                id="DashboardCards__card-value-aria"
              >
                Loading...
              </title>
              <rect
                clip-path="url(#DashboardCards__card-value-diff)"
                height="100%"
                role="presentation"
                style="fill: url(#DashboardCards__card-value-animated-diff);"
                width="100%"
                x="0"
                y="0"
              />
              <defs
                role="presentation"
              >
                <clippath
                  id="DashboardCards__card-value-diff"
                >
                  <rect
                    height="90"
                    width="100%"
                    x="0"
                    y="0"
                  />
                </clippath>
                <lineargradient
                  id="DashboardCards__card-value-animated-diff"
                >
                  <stop
                    offset="0%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-2; -2; 1"
                    />
                  </stop>
                  <stop
                    offset="50%"
                    stop-color="#e5f3fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-1; -1; 2"
                    />
                  </stop>
                  <stop
                    offset="100%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="0; 0; 3"
                    />
                  </stop>
                </lineargradient>
              </defs>
            </svg>
            <span
              class="Typography Typography--label DashboardCards__card-label"
            >
              Activity Risks
            </span>
          </div>
          <div
            class="DashboardCards__card"
          >
            <svg
              aria-labelledby="DashboardCards__card-value-aria"
              class="DashboardCards__card-value"
              height="90"
              role="img"
            >
              <title
                id="DashboardCards__card-value-aria"
              >
                Loading...
              </title>
              <rect
                clip-path="url(#DashboardCards__card-value-diff)"
                height="100%"
                role="presentation"
                style="fill: url(#DashboardCards__card-value-animated-diff);"
                width="100%"
                x="0"
                y="0"
              />
              <defs
                role="presentation"
              >
                <clippath
                  id="DashboardCards__card-value-diff"
                >
                  <rect
                    height="90"
                    width="100%"
                    x="0"
                    y="0"
                  />
                </clippath>
                <lineargradient
                  id="DashboardCards__card-value-animated-diff"
                >
                  <stop
                    offset="0%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-2; -2; 1"
                    />
                  </stop>
                  <stop
                    offset="50%"
                    stop-color="#e5f3fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="-1; -1; 2"
                    />
                  </stop>
                  <stop
                    offset="100%"
                    stop-color="#cce8fe"
                    stop-opacity="1"
                  >
                    <animate
                      attributeName="offset"
                      dur="1.2s"
                      keyTimes="0; 0.25; 1"
                      repeatCount="indefinite"
                      values="0; 0; 3"
                    />
                  </stop>
                </lineargradient>
              </defs>
            </svg>
            <span
              class="Typography Typography--label DashboardCards__card-label"
            >
              Informational Risks
            </span>
          </div>
        </div>
        <div
          class="RiskTypesList"
        >
          <div
            class="RiskTypesList__table"
          >
            <div
              class="RiskTypesList__headers"
            >
              <div
                class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--severity-column"
              >
                Risk severity
              </div>
              <div
                class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--type-column"
              >
                Risk type
              </div>
              <div
                class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--count-column"
              >
                Risk count
              </div>
              <div
                class="RiskTypesList__heading-cell RiskTypesList--actions-column"
              />
            </div>
            <div
              class="RiskTypesList__rows"
            >
              <svg
                aria-labelledby="RiskTypesList__loader-aria"
                class="RiskTypesList__loader"
                height="172"
                role="img"
              >
                <title
                  id="RiskTypesList__loader-aria"
                >
                  Loading...
                </title>
                <rect
                  clip-path="url(#RiskTypesList__loader-diff)"
                  height="100%"
                  role="presentation"
                  style="fill: url(#RiskTypesList__loader-animated-diff);"
                  width="100%"
                  x="0"
                  y="0"
                />
                <defs
                  role="presentation"
                >
                  <clippath
                    id="RiskTypesList__loader-diff"
                  >
                    <rect
                      height="30"
                      width="100%"
                      x="0"
                      y="15"
                    />
                    <rect
                      height="30"
                      width="100%"
                      x="0"
                      y="76"
                    />
                    <rect
                      height="30"
                      width="100%"
                      x="0"
                      y="137"
                    />
                  </clippath>
                  <lineargradient
                    id="RiskTypesList__loader-animated-diff"
                  >
                    <stop
                      offset="0%"
                      stop-color="#eeeeee"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="offset"
                        dur="1.2s"
                        keyTimes="0; 0.25; 1"
                        repeatCount="indefinite"
                        values="-2; -2; 1"
                      />
                    </stop>
                    <stop
                      offset="50%"
                      stop-color="#fdfdfd"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="offset"
                        dur="1.2s"
                        keyTimes="0; 0.25; 1"
                        repeatCount="indefinite"
                        values="-1; -1; 2"
                      />
                    </stop>
                    <stop
                      offset="100%"
                      stop-color="#eeeeee"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="offset"
                        dur="1.2s"
                        keyTimes="0; 0.25; 1"
                        repeatCount="indefinite"
                        values="0; 0; 3"
                      />
                    </stop>
                  </lineargradient>
                </defs>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
